e2e/cypress/integration/integrations/outgoing_webhook/disable_override_username_profile_spec.js

```diff
@@ -0,0 +1,643 @@
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @outgoing_webhook
+/* eslint-disable max-lines */
+
+import * as TIMEOUTS from '../../../fixtures/timeouts';
+
+import {enableUsernameAndIconOverrideInt, enableUsernameAndIconOverride} from '../incoming_webhook/helpers';
+
+describe('Outgoing webhook', () => {
+    let sysadmin;
+    let testTeam;
+    let testChannel;
+    let testUser;
+    let otherUser;
+    let siteName;
+
+    const triggerWord = 'text';
+    const triggerWordExpanded = 'text with some more text';
+    const callbackUrl = 'https://outgoing-webhook-01snptsk6qvz.runkit.sh/';
+    const defaultUserName = 'sysadmin';
+    const noChannelSelectionOption = '--- Select a channel ---';
+    const webhookIconUrlOverride = 'http://via.placeholder.com/150/00F/888';
+    let webhookUserNameOverride = 'webhook';
+    let webhookUserIconOverride = 'webhook_icon.jpg';
+
+    before(() => {
+        cy.apiGetConfig().then(({config}) => {
+            siteName = config.TeamSettings.SiteName;
+        });
+        cy.apiGetMe().then(({user}) => {
+            sysadmin = user;
+        });
+        cy.apiInitSetup().then(({team, channel, user}) => {
+            testTeam = team;
+            testChannel = channel;
+            testUser = user;
+        });
+
+        cy.apiCreateUser().then(({user: user1}) => {
+            otherUser = user1;
+            cy.apiAddUserToTeam(testTeam.id, otherUser.id);
+        });
+    });
+
+    afterEach(() => {
+        cy.apiAdminLogin();
+
+        // # Remove the outgoing webhook
+        // # Go to test team/channel, open main menu and click "Integrations"
+        cy.visit(`${testTeam.name}/channels/${testChannel.name}`);
+        cy.get('#sidebarHeaderDropdownButton', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').click();
+        cy.get('.Menu__content').should('be.visible').findByText('Integrations').click();
+
+        // * click "Outgoing Webhooks"
+        cy.get('.backstage-sidebar').should('be.visible').findByText('Outgoing Webhooks').click();
+
+        // * click "Delete"
+        cy.get('.item-actions > span > button > span').click();
+
+        // * Acknowledge the outgoing webhook deletion
+        cy.get('#confirmModalButton').should('be.visible').click();
+
+        // # Visit integration management at system console and reset override values
+        cy.visit('/admin_console/integrations/integration_management');
+        cy.findByTestId('ServiceSettings.EnablePostUsernameOverride' + false).check({force: true});
+        cy.findByTestId('ServiceSettings.EnablePostIconOverride' + false).check({force: true});
+
+        // # Save the settings
+        cy.get('#saveSetting').scrollIntoView().click({force: true});
+        cy.get('#saveSetting').should('be.disabled');
+    });
```

Comment:
I'd recommend not to do too much UI work in `afterEach`. I've got failing results in other tests and it's not able to recover from it. In short, succeeding tests didn't run. I believe we have a separate test cases for deleting an outgoing webhook and disabling overrides at system console, and so I propose to cover those separately. For test setup related, I'd recommend to do it directly via API.

Instead, I suggests the ff:
```javascript
    before(() => {
        // cy.requireWebhookServer(); // add if https://outgoing-webhook-01snptsk6qvz.runkit.sh/ is added to webhook_serve.js
        cy.apiGetConfig().then(({config}) => {
            siteName = config.TeamSettings.SiteName;
        });
        cy.apiGetMe().then(({user}) => {
            sysadmin = user;
        });
    });

    beforeEach(() => {
        cy.apiAdminLogin();
        cy.apiUpdateConfig({
            ServiceSettings: {
                EnablePostUsernameOverride: false,
                EnablePostIconOverride: false,
            },
        });

        cy.apiInitSetup().then(({team, channel, user}) => {
            testTeam = team;
            testChannel = channel;
            testUser = user;
        });

        cy.apiCreateUser().then(({user: user1}) => {
            otherUser = user1;
            cy.apiAddUserToTeam(testTeam.id, otherUser.id);
        });
    });
```